src/code/z_bgcheck.c

```diff
@@ -1,461 +1,4293 @@
+    func_8003E6C4(&bgActor->dynaLookup);
+    func_8003E6E4(&bgActor->vtxStartIndex);
+    bgActor->boundingSphere.center.x = bgActor->boundingSphere.center.y = bgActor->boundingSphere.center.z = 0;
+    bgActor->boundingSphere.radius = 0;
+}
+
+// setActor internal
+void func_8003E750(BgActor* bgActor, Actor* actor, CollisionHeader* colHeader) {
+    bgActor->actor = actor;
+    bgActor->colHeader = colHeader;
+    bgActor->prevSrp.scale = actor->scale;
+    bgActor->prevSrp.rot = actor->shape.rot;
+    bgActor->prevSrp.rot.x--;
+    bgActor->prevSrp.pos = actor->posRot.pos;
+    bgActor->curSrp.scale = actor->scale;
+    bgActor->curSrp.rot = actor->shape.rot;
+    bgActor->curSrp.pos = actor->posRot.pos;
+}
+
+// Test if BgActor has moved
+s32 func_8003E804(BgActor* bgActor) {
+    return func_8003E5B4(&bgActor->prevSrp, &bgActor->curSrp);
+}
+
+// NULL polyList
+void func_8003E82C(CollisionPoly** polyList) {
+    *polyList = NULL;
+}
+
+// Allocate dyna.polyList
+void func_8003E834(GlobalContext* globalCtx, CollisionPoly** polyList, s32 numPolys) {
+    *polyList = THA_AllocEndAlign(&globalCtx->state.tha, numPolys * sizeof(CollisionPoly), -2);
+    if (!(*polyList != NULL)) {
+        __assert("ptbl->pbuf != NULL", "../z_bgcheck.c", 6247);
+    }
+}
+
+// NULL vtxList
+void func_8003E888(Vec3s** vtxList) {
+    *vtxList = NULL;
+}
+
+// Allocate dyna.vtxList
+void func_8003E890(GlobalContext* globalCtx, Vec3s** vtxList, s32 numVtx) {
+    *vtxList = THA_AllocEndAlign(&globalCtx->state.tha, numVtx * sizeof(Vec3s), -2);
+    if (!(*vtxList != NULL)) {
+        __assert("ptbl->pbuf != NULL", "../z_bgcheck.c", 6277);
+    }
+}
+
+// Update BgActor's prevSrp
+void func_8003E8EC(GlobalContext* globalCtx, BgActor* bgActor) {
+    bgActor->prevSrp = bgActor->curSrp;
+}
+
+// Is BgActor
+s32 func_8003E934(s32 bgId) {
+    if (bgId < 0 || bgId >= BG_ACTOR_MAX) {
+        return false;
+    }
+    return true;
+}
+
+// Init DynaCollisionContext
+void func_8003E954(GlobalContext* globalCtx, DynaCollisionContext* dyna) {
+    dyna->bitFlag = 1;
+    func_8003E82C(&dyna->polyList);
+    func_8003E888(&dyna->vtxList);
+    func_800387FC(globalCtx, &dyna->polyNodes);
+}
+
+// Set DynaCollisionContext
+void func_8003E9A0(GlobalContext* globalCtx, DynaCollisionContext* dyna) {
+    s32 i;
+
+    for (i = 0; i < BG_ACTOR_MAX; i++) {
+        func_8003E6EC(globalCtx, &dyna->bgActors[i]);
+        dyna->bgActorFlags[i] = 0;
+    }
+    func_8003E82C(&dyna->polyList);
+    func_8003E834(globalCtx, &dyna->polyList, dyna->polyListMax);
+
+    func_8003E888(&dyna->vtxList);
+    func_8003E890(globalCtx, &dyna->vtxList, dyna->vtxListMax);
+
+    func_800387FC(globalCtx, &dyna->polyNodes);
+    func_8003880C(globalCtx, &dyna->polyNodes, dyna->polyNodesMax);
+}
+
+// Set BgActor
+// original name: DynaPolyInfo_setActor
+s32 func_8003EA74(GlobalContext* globalCtx, DynaCollisionContext* dyna, Actor* actor, CollisionHeader* colHeader) {
```

Comment:
would be really good if we can name these functions that get used alot in actors. dont know if we want to keep the DynaPolyInfo system name or not, but yea